ui/numbered_menu.py

```diff
@@ -0,0 +1,148 @@
+from threading import Lock
+from time import sleep, time
+
+from ui import Menu
+from ui.utils import clamp, check_value_lock, to_be_foreground
+
+
+class UIWidget(object):
+    pass  # todo : complete
+
+
+class NumberedMenu(UIWidget, Menu):
+    """
+    This subclass of Menu allows the user to navigate using the keypad in the menu. If the menu is 10 entries or less
+    the navigation is instant. If it has more than 10, it let's the user type multiple digits to navigate.
+
+    The `input_delay` parameter controls how long the menu waits before considering an input as definitive.
+    If `input_delay` is 0, then only the 10 first entries are considered navigable with the keypad.
+
+    The `prepend_numbers` parameters controls whether the entries should be prefixed by their number.
+    The default value is `True`
+    """
+    def __init__(self, *args, **kwargs):
+        self.prepend_numbers = kwargs.pop('prepend_numbers') if 'prepend_numbers' in kwargs else True
+        self.input_delay = kwargs.pop('input_delay') if 'input_delay' in kwargs else 1
+        Menu.__init__(self, *args, **kwargs)
+        self.value_lock = Lock()
+        self.custom_key_map = {
+            "KEY_1": 1,
+            "KEY_2": 2,
+            "KEY_3": 3,
+            "KEY_4": 4,
+            "KEY_5": 5,
+            "KEY_6": 6,
+            "KEY_7": 7,
+            "KEY_8": 8,
+            "KEY_9": 9,
+            "KEY_0": 0,
+        }
+        self.entry_count = len(self.contents)
```

Comment:
I suggest making this an attribute. Not only this is something that can be easily expressed as `len(self.contents)` any time it's necessary, it's also going to get outdated as soon as menu contents are re-set with a `menu.set_contents()` call, as `set_contents()` is supposed to be externally available (and is used in both `apps/app_manager.py` and some other apps already)